config_test.go

```diff
@@ -0,0 +1,182 @@
+package cli
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func getFile(t *testing.T) *os.File {
+	tmpFile, err := os.CreateTemp("", "testconfig.yaml")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		os.Remove(tmpFile.Name())
+	})
+	return tmpFile
+}
+
+func TestAddNewConfig(t *testing.T) {
+	tests := []struct {
+		name     string
+		config   func() appConfig
+		expected string
+		errored  bool
+	}{
+		{
+			name: "add first configuration",
+			config: func() appConfig {
+				cfg := newDefaultConfig()
+				cfg.Name = "BestConfig"
+				cfg.AccessKey = "FamousKey"
+				cfg.AccessSecretKey = "TopSecret"
+				return cfg
+			},
+			expected: `BestConfig:
+    access-key: FamousKey
+    access-secret-key: TopSecret
+    url: https://chat.stream-io-api.com
+    default: true
+`,
+		},
+		{
+			name: "add second configuration",
+			config: func() appConfig {
+				cfg := newDefaultConfig()
+				cfg.Name = "BestConfigEver"
+				cfg.AccessKey = "FamousKey"
+				cfg.AccessSecretKey = "TopSecret"
+				return cfg
+			},
+			expected: `BestConfig:
+    access-key: FamousKey
+    access-secret-key: TopSecret
+    url: https://chat.stream-io-api.com
+    default: true
+BestConfigEver:
+    access-key: FamousKey
+    access-secret-key: TopSecret
+    url: https://chat.stream-io-api.com
+`,
+		},
+		{
+			name: "add already existing configuration",
+			config: func() appConfig {
+				cfg := newDefaultConfig()
+				cfg.Name = "BestConfig"
+				cfg.AccessKey = "FamousKey"
+				cfg.AccessSecretKey = "TopSecret"
+				return cfg
+			},
+			errored: true,
+		},
+	}
+
+	file := getFile(t)
+	config := &Config{
+		filepath: file.Name(),
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			conf := test.config()
+			err := config.Add(conf)
+			if test.errored {
+				require.Error(t, err)
+				return
+			}
+
+			content, err := ioutil.ReadFile(file.Name())
```

Comment:
ioutil is being considered as deprecated, so good to use os package instead

Reply:
bad habit